go-controller/pkg/ovn/pods.go

```diff
@@ -115,7 +115,13 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod, portInfo *lpInfo) (err er
 		// is not re-added into the cache. Delete logical switch port anyway.
 		annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
 		if err != nil {
-			return fmt.Errorf("unable to unmarshal pod annocations for pod %s/%s: %w", pod.Namespace, pod.Name, err)
+			if util.IsAnnotationNotSetError(err) {
```

Comment:
@flavio-fernandes : can you please add a unit test in pods_test.go to capture this case of triggering deletion on a pod which doesn't have an annotation set?